src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateGeneratorTests.cs

```diff
@@ -291,13 +276,18 @@ public async Task MapAction_NoParam_AnyReturn(string source, string expectedBody
         await VerifyResponseBodyAsync(httpContext, expectedBody);
     }
 
-    [Theory]
-    [InlineData(@"app.MapGet(""/"", () => new Todo() { Name = ""Test Item""});")]
-    [InlineData("""
-object GetTodo() => new Todo() { Name = "Test Item"};
+    public static IEnumerable<object[]> MapAction_NoParam_ComplexReturn_Data => new List<object[]>()
+    {
+        new object[] { $$"""app.MapGet("/", () => new {{typeof(Todo)}}() { Name = "Test Item"});""" },
```

Comment:
Similar comment here about needing to use `typeof`. If `Todo` was ever renamed, this code would fail to compile and the test would fail. I think it would be safe (and more readable) to just hard-code `Todo` here.

```suggestion
        new object[] { """app.MapGet("/", () => new Todo() { Name = "Test Item"});""" },
```

Reply:
We need to use the same `Todo` type in the compiled code that exists in the test code. Previously, the test code had its own `Todo` type defined that happened to have the same properties as the `Todo` type in the test code with differing nullability.

This is incorrect because we end up instantiated one type in test code and using another in the actual endpoint. Using the same type is more consistent and doesn't require us to rely on properties incidentally being deserialized the same way.